pkg/utils/installer/driver-deployment.go

```diff
@@ -0,0 +1,180 @@
+	podSpec := corev1.PodSpec{
+		ServiceAccountName: name,
+		Volumes: []corev1.Volume{
+			newHostPathVolume(volumeNameSocketDir,
+				newDirectCSIPluginsSocketDir(kubeletDirPath, fmt.Sprintf("%s-controller", name)),
+			),
+			newSecretVolume(admissionControllerCertsDir, AdmissionWebhookSecretName),
+			newSecretVolume(conversionWebhookCertVolume, conversionWebhookCertsSecret),
+		},
+		Containers: []corev1.Container{
+			{
+				Name:  csiProvisionerContainerName,
+				Image: filepath.Join(registry, org, csiProvisionerContainerImage),
+				Args: []string{
+					fmt.Sprintf("--v=%d", logLevel),
+					"--timeout=300s",
+					fmt.Sprintf("--csi-address=$(%s)", endpointEnvVarCSI),
+					"--leader-election",
+					"--feature-gates=Topology=true",
+					"--strict-topology",
+				},
+				Env: []corev1.EnvVar{
+					{
+						Name:  endpointEnvVarCSI,
+						Value: "unix:///csi/csi.sock",
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{
+					newVolumeMount(volumeNameSocketDir, "/csi", false),
+				},
+				TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+				TerminationMessagePath:   "/var/log/controller-provisioner-termination-log",
+				// TODO: Enable this after verification
+				// LivenessProbe: &corev1.Probe{
+				// 	FailureThreshold:    5,
+				// 	InitialDelaySeconds: 10,
+				// 	TimeoutSeconds:      3,
+				// 	PeriodSeconds:       2,
+				// 	Handler: corev1.Handler{
+				// 		HTTPGet: &corev1.HTTPGetAction{
+				// 			Path: healthZContainerPortPath,
+				// 			Port: intstr.FromInt(9898),
+				// 		},
+				// 	},
+				// },
+				SecurityContext: &corev1.SecurityContext{
+					Privileged: &privileged,
+				},
+			},
+			{
+				Name:  directCSIContainerName,
+				Image: filepath.Join(registry, org, directCSIContainerImage),
+				Args: []string{
+					fmt.Sprintf("--v=%d", logLevel),
+					fmt.Sprintf("--identity=%s", name),
+					fmt.Sprintf("--endpoint=$(%s)", endpointEnvVarCSI),
+					fmt.Sprintf("--conversion-webhook-url=%s", conversionWebhookURL),
+					"--controller",
+				},
+				SecurityContext: &corev1.SecurityContext{
+					Privileged: &privileged,
+				},
+				Ports: []corev1.ContainerPort{
+					{
+						ContainerPort: admissionControllerWebhookPort,
+						Name:          admissionControllerWebhookName,
+						Protocol:      corev1.ProtocolTCP,
+					},
+					{
+						ContainerPort: 9898,
+						Name:          "healthz",
+						Protocol:      corev1.ProtocolTCP,
+					},
+				},
+				Env: []corev1.EnvVar{
+					{
+						Name: kubeNodeNameEnvVar,
+						ValueFrom: &corev1.EnvVarSource{
+							FieldRef: &corev1.ObjectFieldSelector{
+								APIVersion: "v1",
+								FieldPath:  "spec.nodeName",
+							},
+						},
+					},
+					{
+						Name:  endpointEnvVarCSI,
+						Value: "unix:///csi/csi.sock",
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{
+					newVolumeMount(volumeNameSocketDir, "/csi", false),
+					newVolumeMount(admissionControllerCertsDir, certsDir, false),
```

Comment:
Can we mount the admission controller certs only when `admission-control` flag is enabled?